ndk/hooks/useFeedWithEose.ts

```diff
@@ -0,0 +1,39 @@
+import { type NDKEvent, type NDKFilter } from "@nostr-dev-kit/ndk";
+import { useNDK } from "ndk/NDKProvider";
+import { createRelaySet } from "ndk/utils";
+import { useEffect, useRef, useState } from "react";
+
+export function useFeedWithEose(filter: NDKFilter, relayUrls: string[] = []) {
+  const { ndk } = useNDK();
+  const [feed, setFeed] = useState<NDKEvent[]>([]);
+  const eventBatch = useRef<NDKEvent[] | null>([]);
```

Comment:
This doesn't need the null option right?

Reply:
i set it to null after eose. i might get rid of this hook though. i got another idea after thinking about this more.